erts/emulator/beam/erl_bif_lists.c

```diff
@@ -906,7 +947,10 @@ static int subtract_continue(Process *p, ErtsSubtractContext *context) {
                 return res;
             }
 
-            if (context->lhs_remaining <= SUBTRACT_LHS_THRESHOLD) {
+            /* If the lhs list is empty then there's nothing to do.
+             * Returning early will be taken care of in the `SUBTRACT_STAGE_LEN_RHS`
+             * stage after the rhs list has been scanned. */
+            if (context->lhs_remaining > 0 && context->lhs_remaining <= SUBTRACT_LHS_THRESHOLD) {
```

Comment:
We can skip rebuilds of small lists by checking whether RHS =/= [] here.

```suggestion
            if (context->lhs_remaining > 0 &&
                context->lhs_remaining <= SUBTRACT_LHS_THRESHOLD &&
                is_list(context->rhs_original)) {
```